src/object/bumper.cpp

```diff
@@ -60,7 +64,33 @@ Bumper::update(float dt_sec)
 {
   if (m_sprite->animation_done())
     set_action("normal", m_dir);
-  m_col.set_movement(m_physic.get_movement (dt_sec));
+
+  // Pushing rocks, as well as dynamic with tilemap, platform, and fallblock.
+
+  Rectf small_overlap_box = get_bbox().grown(1.f);
+
+  for (auto& rock : Sector::get().get_objects_by_type<Rock>())
+  {
+    if (small_overlap_box.overlaps(rock.get_bbox()))
+    {
+      float BOUNCE_DIR = m_dir == Direction::LEFT ? -BOUNCE_X : BOUNCE_X;
+      rock.get_physic().set_velocity(BOUNCE_DIR * 0.7f, BOUNCE_Y * 0.6f);
+      SoundManager::current()->play(TRAMPOLINE_SOUND, get_pos());
+      m_sprite->set_action("swinging", m_dir, 1);
```

Comment:
Call this directly on `this` for a collision-safe approach (applies to `MovingSprite`s):

```suggestion
      set_action("swinging", m_dir, 1);
```